beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -314,8 +311,36 @@ async fn compute_and_publish_data_columns<T: BeaconChainTypes>(
                     })
                     .map_err(FetchEngineBlobError::DataColumnSidecarError)?;
 
-                publish_fn(BlobsOrDataColumns::DataColumns(custody_columns.clone()));
-                Ok(custody_columns)
+                // Gossip verify data columns before publishing. This prevents blobs with invalid
+                // KZG proofs from the EL making it into the data availability checker. We do not
+                // immediately add these blobs to the observed blobs/columns cache because we want
+                // to allow blobs/columns to arrive on gossip and be accepted (and propagated) while
+                // we are waiting to publish. Just before publishing we will observe the blobs/columns
+                // and only proceed with publishing if they are not yet seen.
+                let columns_to_import_and_publish = custody_columns
+                    .into_iter()
+                    .filter_map(|col| {
+                        let col_index = col.index;
+                        match GossipVerifiedDataColumn::<T, DoNotObserve>::new(
+                            col,
+                            col_index,
+                            &chain_cloned,
+                        ) {
+                            Ok(verified) => Some(Ok(verified)),
+                            // Ignore already seen data columns
+                            Err(GossipDataColumnError::PriorKnown { .. })
+                            | Err(GossipDataColumnError::PriorKnownUnpublished) => None,
```

Comment:
can we match all errors here? if there's another error variant we add that we need to potentially ignore later, then it will take forever to find the bug with fetch_blobs

Reply:
I generally agree with this, although in this case `PriorKnown`s are kind of like the exception here and we don't do any handling for other error variants, and just let errors propagate up. I imagine for new variants we'd want to do the same thing?

Reply:
I don't have a strong preference on this though, happy to make the change - but just wanted to share my thoughts.

Reply:
Addressed in 966dadae6f00f0e363953d42d6861e25388c7a2b

Reply:
Addressed in 966dadae6f00f0e363953d42d6861e25388c7a2b